ring/replication_set_tracker.go

```diff
@@ -505,7 +512,14 @@ func (t *inflightInstanceTracker) removeInstance(replicationSetIdx int, instance
 	t.mx.Lock()
 	defer t.mx.Unlock()
 
-	delete(t.inflight[replicationSetIdx], instance)
+	for i, instances := 0, t.inflight[replicationSetIdx]; i < len(instances); i++ {
```

Comment:
will a `range` work here too?

Reply:
Yes, we can. No impact on perf. Done in https://github.com/grafana/dskit/pull/498/commits/4cd5dbbe9ccebb0097799bb29d29ae929cc0de31